src/utils/generator.js

```diff
@@ -197,6 +197,21 @@ export class Generator {
     );
   }
 
+  emitDoWhileStatement(test: AbstractValue, body: Generator) {
+    this._addEntry({
+      args: [],
+      buildNode: function([], context: SerializationContext) {
+        let statements = context.serializeGenerator(body);
+        let testId = test.intrinsicName;
+        invariant(testId !== undefined);
+        statements.push(t.ifStatement(t.identifier(testId), t.continueStatement(), t.breakStatement()));
```

Comment:
What's wrong with emitting the condition as part of the `while` clause? A comment would be nice if there's a reason.

Reply:
I'll look at this again.

Reply:
I'll move it into the while clause.